edalize/rivierapro.py

```diff
@@ -84,7 +90,20 @@ def _write_build_rtl_tcl_file(self, tcl_main):
                 args += ['-quiet']
                 args += ['-work', f.logical_name]
                 args += [f.name.replace('\\','/')]
-                tcl_build_rtl.write("{} {}\n".format(cmd, ' '.join(args)))
+                common_compilation += [f.name.replace('\\','/'),'\\\n']
+
+                if (self.tool_options.get('compilation_mode'))==['sep']:
+                    tcl_build_rtl.write("{} {}\n".format(cmd, ' '.join(args)))
+
+        if (self.tool_options.get('compilation_mode')==['common']) or (self.tool_options.get('compilation_mode')==None):
+            tcl_build_rtl.write("{} \n".format(' '.join(common_compilation)))
+			
+        if not (self.tool_options.get('compilation_mode')==['common'] or self.tool_options.get('compilation_mode')==None or self.tool_options.get('compilation_mode')==['sep']):
+            print('wrong compilation mode, use --compilation_mode=common for common compilation or --compilation_mode=sep for separate compilation')
+            sys.exit()
+
```

Comment:
Instead of print and exit we should raise a `RuntimeError` when we encounter illegal options. You can check how it's done in the verilator backend for handling the `mode` option for reference https://github.com/olofk/edalize/blob/master/edalize/verilator.py#L86-L97